iroh-gateway/src/error.rs

```diff
@@ -33,10 +33,14 @@ impl GatewayError {
 impl IntoResponse for GatewayError {
     fn into_response(self) -> Response {
         let mut headers = HeaderMap::new();
-        headers.insert(
-            &HEADER_X_TRACE_ID,
-            HeaderValue::from_str(&self.trace_id).unwrap(),
-        );
+        let has_trace_id =
+            !self.trace_id.is_empty() && self.trace_id != "00000000000000000000000000000000";
```

Comment:
So why is this a string? We don't want to expose the TraceId type from OpenTelemetry?

It seems that TraceId has a public API to convert it to and from [u8; 16]. Maybe use that for traceid? Then we don't have to do string comparisons here.

Reply:
Only reason was not to pull opentelemetry directly into iroh-gateway cause at some point we wanted to have metrics be a compile time toggle. I guess no hard reason right now. And the dep can be bound to the feature anyways.

I'll fix it up before merge.

Reply:
Also we still have the dependency anyways so duh...
Nice catch ty.